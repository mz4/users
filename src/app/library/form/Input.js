import React from 'react';
import { Controller } from 'react-hook-form';
import styles from './Input.module.scss';

export const Input = ({
  control,
  defaultValue,
  placeholder,
  name,
  errors,
  disabled,
  dataTestId,
  type = 'text',
  addEmptySpace = true,
}) => {
  return (
    <div className={styles.inputContainer}>
      <Controller
        name={name}
        control={control}
        defaultValue={defaultValue || ''}
        render={({ field: { onChange, onBlur, value } }) => (
          <input
            disabled={disabled}
            placeholder={placeholder}
            type={type}
            value={value}
            onChange={(evt) => {
              return onChange(evt);
            }}
            onBlur={onBlur}
            data-testid={dataTestId}
          />
        )}
      />
      {errors[name] ? (
        <div className={styles.error}>{errors[name].message}</div>
      ) : (
        addEmptySpace && <div className={styles.emptyError} />
      )}
    </div>
  );
};
