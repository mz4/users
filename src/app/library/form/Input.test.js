import renderer from 'react-test-renderer';
import { useForm } from 'react-hook-form';
import { render } from '@testing-library/react';
import { renderHook, act } from '@testing-library/react-hooks';
import { Input } from './Input';

describe('Render field component', () => {
  const { result } = renderHook(() =>
    useForm({
      testname: 'one'
    })
  );
  expect(result.current).toBeDefined();
  result.current.setValue('testname', 'one');
  result.current.getValues();
  const field = (
    <Input
      control={result.current.control}
      defaultValue={'twllllo'}
      placeholder={'testplaceholder'}
      name={'testname'}
      errors={[]}
      disabled={false}
    />
  );
  it('Create a snapshot', () => {
    const component = renderer.create(field);
    expect(component.toJSON()).toMatchSnapshot();
  });
  it('Invoke onChange', () => {
    act(() => {
      result.current.setValue('testname', 'two');
    });
    const { getByPlaceholderText } = render(field);
    const input = getByPlaceholderText('testplaceholder', { name: 'testname' });
    expect(input.value).toBe('two');
    expect(input).toBeEnabled();
  });
});
