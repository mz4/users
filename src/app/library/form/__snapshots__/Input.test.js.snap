// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Render field component Create a snapshot 1`] = `
<div
  className="inputContainer"
>
  <input
    disabled={false}
    onBlur={[Function]}
    onChange={[Function]}
    placeholder="testplaceholder"
    type="text"
    value="one"
  />
  <div
    className="emptyError"
  />
</div>
`;
